server/src/main/java/fr/uga/l3miage/tp4/repositories/CandidateEntityRepository.java

```diff
@@ -0,0 +1,20 @@
+package fr.uga.l3miage.tp4.repositories;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
+import fr.uga.l3miage.tp4.models.CandidateEntity;
+
+import java.time.LocalDate;
+import java.util.Set;
+
+@Repository
+public interface CandidateEntityRepository extends JpaRepository<CandidateEntity, Long> {
+        //ilya une erreur ici qui empeche le projet de build du coup je lai mis en commentaire pour que le projet build
+
+        //  Set<CandidateEntity> findByTestCenterCode(String code);
```

Comment:
Et oui logique car ici votre requête est bonne mais vous ne donnez pas le bon paramètre à la fonction, un `TestCenterCode` , n'est pas un string ! 
```suggestion
        Set<CandidateEntity> findAllByTestCenterCode(TestCenterCode code);
```